spacepy/toolbox/__init__.py

```diff
@@ -392,8 +386,6 @@ def loadpickle(fln):
                 return pickle.load(fh, encoding='latin1')
         except pickle.UnpicklingError: #maybe it's a gzip?
             gzip = True
-        else:
-            gzip = False
     if gzip:
```

Comment:
I think the idea was we don't even need this "if"...gzip is guaranteed to be true if we reach this point.

- the .gz exists and fln doesn't, we reach this point and gzip is True
- the fln exists OR the fln doesn't: we try to load the pickle. If it succeeds, return out. If it fails, gzip is set True. (Or some other exception is raised, e.g. file not existing....)

So we should just rip out the `if gzip` and dedent that block.

Reply:
Ah, I see. I think we can actually cut out the gzip logic completely - we don't use the `gzip` boolean outside of line 389, and as you explained there's no way for `gzip` to be false at that point (or ever). The occurrences of `gzip = True` are pretty illustrative of the intention of the code, I'll put some comments in so that's not lost.